machine-learning/app/config.py

```diff
@@ -25,30 +25,20 @@ class FacialRecognitionSettings(BaseModel):
 
 
 class PreloadModelData(BaseModel):
-    clip: ClipSettings = ClipSettings()
-    facial_recognition: FacialRecognitionSettings = FacialRecognitionSettings()
-
-    clip_model_fallback: str | None = os.getenv("MACHINE_LEARNING_PRELOAD__CLIP", None)
-    facial_recognition_model_fallback: str | None = os.getenv("MACHINE_LEARNING_PRELOAD__FACIAL_RECOGNITION", None)
-
-    def update_from_fallbacks(self) -> None:
-        if self.clip_model_fallback:
-            self.clip.textual = self.clip_model_fallback
-            self.clip.visual = self.clip_model_fallback
-            log.warning(
-                "Deprecated env variable: MACHINE_LEARNING_PRELOAD__CLIP. "
-                "Use MACHINE_LEARNING_PRELOAD__CLIP__TEXTUAL and "
-                "MACHINE_LEARNING_PRELOAD__CLIP__VISUAL instead."
-            )
-
-        if self.facial_recognition_model_fallback:
-            self.facial_recognition.recognition = self.facial_recognition_model_fallback
-            self.facial_recognition.detection = self.facial_recognition_model_fallback
-            log.warning(
-                "Deprecated environment variable: MACHINE_LEARNING_PRELOAD__FACIAL_RECOGNITION. "
-                "Use MACHINE_LEARNING_PRELOAD__FACIAL_RECOGNITION__RECOGNITION and "
-                "MACHINE_LEARNING_PRELOAD__FACIAL_RECOGNITION__DETECTION instead."
-            )
+    clip: ClipSettings | str = ClipSettings()
+    facial_recognition: FacialRecognitionSettings | str = FacialRecognitionSettings()
+    clip_fallback: str | None = None
+    facial_recognition_fallback: str | None = None
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        if isinstance(self.clip, str):
+            self.clip = ClipSettings(textual=self.clip, visual=self.clip)
+            self.clip_fallback = "True"
+        if isinstance(self.facial_recognition, str):
+            self.clip = ClipSettings(textual=self.facial_recognition, visual=self.facial_recognition)
+            self.facial_recognition_fallback = "True"
```

Comment:
```suggestion
        self.clip_fallback = isinstance(self.clip, str);
        if self.clip_fallback:
            self.clip = ClipSettings(textual=self.clip, visual=self.clip)

        self.facial_recognition_fallback = isinstance(self.facial_recognition, str);
        if self.facial_recognition_fallback:
            self.facial_recognition = FacialRecognitionSettings(detection=self.facial_recognition, recognition=self.facial_recognition)
```
Make the fallbacks `bool` instead.

Reply:
Did you change the types to `bool`?

Reply:
You can remove `str` from the type with something like `assert not isinstance(settings.clip, str)` or `assert isinstance(settings.clip, ClipSettings`.

Reply:
Is there a particular reason you changed the `clip` and `facial_recognition` fields to be union types?

Reply:
Just handle the deprecated envs in `preload_models` with `os.getenv` without all of this messy stuff. It's waaay over-complicating this.